concourse/pipelines/linux-image-build.yaml

```diff
@@ -17,6 +17,12 @@ resources:
   source:
     uri: https://github.com/GoogleCloudPlatform/guest-test-infra.git
     branch: master
+- name: imagetest
+  type: git
+  source:
+    uri: https://github.com/GoogleCloudPlatform/guest-test-infra.git
+    branch: master
+    paths: ['imagetest/.*\.go']
```

Comment:
So this is not a new repo, it's the subfolder and concourse will only download the subfolder of the repo.